python_modules/dagster/dagster/_core/definitions/declarative_automation/serialized_objects.py

```diff
@@ -43,15 +48,48 @@ class HistoricalAllPartitionsSubsetSentinel:
     """
 
 
+def _get_maybe_compressed_dynamic_partitions_subset(
+    subset: EntitySubset,
+) -> SerializableEntitySubset:
+    # for DefaultPartitionsSubset on a DynamicPartitionsDefinition, we convert this to a
+    # KeyRangesPartitionsSubset. this is technically a lossy conversion as it's possible
+    # for the set of keys between the start and end of a range to change after serialization
+    # (e.g. if a partition is deleted). however, accuracy to that degree is not necessary
```

Comment:
a partition being deleted should be fine since we still validate against get_partition_keys() right? the key getting deleted and re-added is what sounds potentially a bit spicier, but as long as it isn't permanently excluded and is treated similarly to the way a brand new partition key would, probably ok?

Reply:
yep that's accurate, updated the comment